src/lib.rs

```diff
@@ -94,35 +111,132 @@ where <T as SmartLedsWrite>::Color: From<RGB8> {
 /// The map() function shall fix any x y coordinate mismatch. Mismatch means
 /// the matrix might display the result being drawn in mirrored or otherwise
 /// incorrect ways due to the LEDs order on the PCB.
-/// The size() function returns the x and y size of the matrix to satisfy
-/// embedded-graphics user libraries.
-pub trait MatrixType {
-    fn map(&self, pos: Point) -> Result<Point, DisplayError>;
-    fn size(&self) -> Size;
+/// Grid type matrixes (like 2x2  of 1x4 grid of 8x8 matrixes) should be also
+/// handled using this trait.
+pub trait Transformation<const W: usize, const H: usize> {
+    fn map(pos: Point) -> Option<Point>;
 }
 
-/// Type definition for simple 8x8 matrix.
-pub struct MT8x8 {
+/// Transformation to fix Y inversion on certain matrixes.
+pub enum InvertY {}
+
+/// No-op transformation.
+pub enum None {}
```

Comment:
This is a bit nitpicky, but could the name `None` be confusing in some situations, because `None` is also used for `Option::None`? Perhaps you could instead use `Identity`.